src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/StringBuilderAppendShouldNotTakeSubstring.Fixer.cs

```diff
@@ -0,0 +1,173 @@
+                                equivalenceKey: MicrosoftApiDesignGuidelinesAnalyzersResources.StringBuilderShouldUseSubstringOverloadWithTwoParameterFix),
+                            diagnostic);
+                }
+            }
+        }
+
+        private static async Task<Document> FixCodeOneParameter(
+            Document document,
+            SyntaxNode root,
+            SyntaxNode nodeToFix,
+            CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var typedNodeToFix =  semanticModel.GetOperation(nodeToFix) as IInvocationOperation;
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var fixComponents = GetFixComponents(generator, typedNodeToFix);
+
+            SyntaxNode lengthNode = generator
+                .MemberAccessExpression(
+                    fixComponents.StringArgument.Syntax, 
+                    generator.IdentifierName(nameof(string.Length)));
+
+            var startIndexNode = fixComponents.OriginalInnerArguments[0].Value.Syntax;
+
+            var lengthArgument = generator
+                .SubtractExpression(
+                    lengthNode,
+                    startIndexNode);
+
+            var newNode = generator.InvocationExpression(
+                fixComponents.TargetMethod, 
+                fixComponents.StringArgument.Syntax,
+                startIndexNode, 
+                lengthArgument);
+
+            var newRoot = root
+                .ReplaceNode(
+                    nodeToFix, 
+                    newNode);
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private struct FixComponents
+        {
+            public FixComponents(
+                IOperation stringArgument, 
+                SyntaxNode targetMethod, 
+                ImmutableArray<IArgumentOperation> originalInnerArguments)
+            {
+                StringArgument = stringArgument;
+                TargetMethod = targetMethod;
+                OriginalInnerArguments = originalInnerArguments;
+            }
+
+            public IOperation StringArgument { get; }
+            public SyntaxNode TargetMethod { get; }
+            public ImmutableArray<IArgumentOperation> OriginalInnerArguments { get; }
+        }
+
+        private static async Task<Document> FixCodeTwoParameters(
+            Document document,
+            SyntaxNode root,
+            SyntaxNode nodeToFix,
+            CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var typedNodeToFix = semanticModel.GetOperation(nodeToFix) as IInvocationOperation;
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            FixComponents fixComponents = GetFixComponents(generator, typedNodeToFix);
```

Comment:
the mixing of 'var' and non-var all over the place is pretty confusing TBH.  can you just switch to 'var' everywhere?